shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionFeature.kt

```diff
@@ -42,12 +42,13 @@ interface StepCompletionFeature {
          */
 
         sealed interface CheckTopicCompletionStatus : Message {
-            data class Completed(val modalText: String) : CheckTopicCompletionStatus
+            data class Completed(val modalText: String, val nextStepId: Long?) : CheckTopicCompletionStatus
             object Uncompleted : CheckTopicCompletionStatus
             object Error : CheckTopicCompletionStatus
         }
 
         object TopicCompletedModalGoToHomeScreenClicked : Message
+        data class TopicCompletedModalContinueNextTopicClicked(val nextStepRoute: StepRoute) : Message
```

Comment:
In my opinion it's better not to pass nextStepRoute to the platform, but store it in the state, and then in reducer create `StepRoute` when handling `TopicCompletedModalContinueNextTopicClicked` message.